src/TwigComponent/src/ComponentFactory.php

```diff
@@ -30,11 +31,12 @@ final class ComponentFactory
      * @param array<class-string, string> $classMap
      */
     public function __construct(
+        private Environment $environment,
```

Comment:
WDYT about passing something smaller here - a new `ComponentTemplateFinderInterface` - with a single method on it.

Reply:
It's a good idea and that let more flexibility for people to overwrite and structure differently their project